tools/src/wyvern/target/corewyvernIL/type/StructuralType.java

```diff
@@ -101,7 +101,7 @@ public boolean isSubtypeOf(ValueType t, TypeContext ctx) {
 		if (t instanceof NominalType) {
 			StructuralType st = ((NominalType) t).getStructuralType(ctx, null);
 			if (st == null) 
-				return false; // abstract type; I am not a subtype of this
+				return true; // abstract type; I am always a subtype of this
```

Comment:
Yes, it does. As far as I can tell, the only reason why the test fails is because this always returns false, even when the types are matching. Perhaps a more sophisticated mechanism would be better?
I don't understand why any type shouldn't be a subtype of an abstract type.